react-delegationdashboard/src/contracts/Delegation.tsx

```diff
@@ -70,9 +70,14 @@ export default class Delegation {
       throw new Error('The contract for this action in not defined');
     } else {
       let funcName = delegationContract.data;
+      let gasLimit = delegationContract.gasLimit;
       if (delegationTransactionType.args !== '') {
         funcName = `${delegationContract.data}${delegationTransactionType.args}`;
       }
+      if (delegationContract.data === 'addNodes' && delegationTransactionType.args) {
+        const nodesNo = delegationTransactionType.args.split('@').slice(1).length / 2;
```

Comment:
might rename to `numNodes`.
Also there are a lot of operations done in this line. maybe split it in multiple lines?